src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/EETypeNode.cs

```diff
@@ -86,7 +86,7 @@ public EETypeNode(NodeFactory factory, TypeDesc type)
         public override bool ShouldSkipEmittingObjectNode(NodeFactory factory)
         {
             // If there is a constructed version of this node in the graph, emit that instead
-                if (ConstructedEETypeNode.CreationAllowed(_type))
+            if (ConstructedEETypeNode.CreationAllowed(_type) || _type.IsCanonicalDefinitionType(CanonicalFormKind.Any))
```

Comment:
If you are making it possible for canonical definition EEType nodes to be "constructed", you should update `ConstructedEETypeNode.CreationAllowed`, not make a change that makes some particular problem you had disappear. Not updating `CreationAllowed` will likely lead to other problems down the line. `CreationAllowed` specifically means "can I get a fully loaded EEType for this thing"? With your change, you can.